soroban-sdk/src/deploy.rs

```diff
@@ -22,7 +22,13 @@
 //! #     pub fn f(env: Env, wasm_hash: BytesN<32>) {
 //! #         let salt = [0u8; 32];
 //! #         let deployer = env.deployer().with_current_contract(salt);
-//! #         let contract_address = deployer.deploy(wasm_hash);
+//! #         // Deployed contract address is deterministic and can be accessed
+//! #         // without actually deploying the contract.
+//! #         let computed_deployed_address = deployer.deployed_address();
+//! #         let deployed_address = deployer.deploy(wasm_hash);
+//! #         if computed_deployed_address != deployed_address {
+//! #             panic!("computed deployed contract address has to be consistent with actual deployed address");
```

Comment:
Can we reduce this example as follows?
```suggestion
//! #         // Deployed contract address is deterministic and can be accessed
//! #         // before deploying the contract.
//! #         let _ = deployer.deployed_address();
```

I think including a comparison test in the example could be confusing to someone. Examples are for copying and pasting, and I don't think we hope folks will do the comparison themselves. Someone could interpret this to mean they need to compare it before and after deploy.

That would be a good thing to chuck in a test though in the sdk/tests folder.

Reply:
Yeah, we don't have great test coverage of this whole file for now (asset tests are needed as well), so probably tests should be done separately as a followup.